### DISCUSSION TITLE
How to allow cross-origin iframe access for WebKit/Playwright?

### DISCUSSION THREAD

**Discussion by Clement398** [cypress-io/cypress#24789] [Questions and Help] [answered]
Hi, do you by any chance know if there is an equivalent to chromeWebSecurity: false for WebKit/Playwright to be able to access a cross-origin iframe (Stripe credit card)?

**Comment by portyu9** [comment, marked as answer]
Hi @Clement398 

There is no Playwright/WebKit equivalent of Cypress's `chromeWebSecurity: false` that you need for this.

Playwright can interact with elements inside **cross-origin iframes directly** through its frame APIs. You should locate the Stripe iframe and then locate the field inside that frame:

```js
const cardFrame = page.frameLocator('iframe[title="..."]')

await cardFrame.getByLabel('Card number').fill('4242424242424242')
```

Or with the newer locator API:

```js
const cardFrame = page.locator('iframe[title="..."]').contentFrame()

await cardFrame.getByLabel('Card number').fill('4242424242424242')
```

Use a selector specific enough to identify the correct Stripe frame. Stripe integrations can also create separate frames for different payment fields, so don't assume there is only one `iframe`.

What will fail is trying to access the frame through normal in-page JavaScript such as:

```js
iframe.contentDocument
```

because the browser's same-origin policy applies to application JavaScript.

Playwright's automation API handles frame interaction outside that page-JavaScript restriction, so disabling browser web security should not be necessary.

Please mark this answer as accepted if it helped, thank you.
